File: ts/src/types/set.ts
/**
 * Set operation request types
 */

/**
 * Move set member request
 */
export interface MoveSetMemberRequest {
  member: string;
  destination: string;
}

/**
 * Set operation request (union, intersection, difference)
 */
export interface SetOperationRequest {
  keys: string[];
}

/**
 * Batch set members request
 */
export interface BatchSetMembersRequest {
  [key: string]: string[];
}

/**
 * Set operation types
 */

export interface SetOperation {
  key: string;
  member: string;
}

File: crates/redis_client/src/common/set.rs
use serde::{ Deserialize, Serialize };

/// Request for adding a member to a set
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SetMemberRequest {
    pub member: String,
}

/// Request for adding multiple members to a set
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SetMembersRequest {
    pub members: Vec<String>,
}

/// Request for set operations (intersect, union, difference)
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SetKeysRequest {
    pub keys: Vec<String>,
}

/// Set operation for batch operations
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SetOperation {
    pub key: String,
    pub members: Vec<String>,
}

/// Set information response
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SetInfo {
    pub key: String,
    pub members: Vec<String>,
    pub cardinality: usize,
    pub ttl: Option<i64>,
}
